tensorboard/plugins/projector/vz_projector/data.ts

```diff
@@ -444,9 +443,14 @@ export class DataSet {
   private async computeKnn(
     data: DataPoint[], 
     nNeighbors: number): Promise<knn.NearestEntry[][]> {
-    if (this.nearest != null && nNeighbors <= this.nearest.length) {
-      // We found the nearest neighbors before and will reuse them.
-      return Promise.resolve(this.nearest);
+    
+    // Handle the case where we've previously found the nearest neighbors.
+    const previouslyComputedNNeighbors = this.nearest ? 
+      this.nearest[0].length : 0;
```

Comment:
In the edge case that the dataset is empty, this will crash with “Cannot
read property `length` of `undefined`”, where the correct behavior is
presumably to return `[]`.

Reply:
Will `result` computed from `knn.findKNN` exist? There's a ton of `any` types floating around here, which makes it hard to reason about what the state of this  value ought to be. Anyway, I think I've changed it such that this edge issue won't sink things.

Reply:
> Will result computed from `knn.findKNN` exist?

I’d assume that it would exist and equal `[]`, and from a glance at the
source this looks accurate (but I could be wrong). In any case, your
patch looks fine; thanks!